### PR TITLE
SC Substation Conns Check

### PR DESCRIPTION
**Pull request by MRossol**
Add a verbose substations connection to trans line check
Refactor Supply Curve to run lcot calc outside of the init and handle substation connections only in full sort

### CODE REVIEW

File: reV/supply_curve/supply_curve.py [hunk 1]
```diff
@@ -295,7 +253,63 @@ def _map_trans_capacity(trans_sc_table):
...
+        """
+        if isinstance(trans_line_gids, str):
+            trans_line_gids = json.loads(trans_line_gids)
+
+        return trans_line_gids
+
+    @classmethod
+    def _check_sub_trans_lines(cls, features):
+        """
+        Check to make sure all trans-lines are available for all sub-stations
```
Comment (grantbuster) [inline reV/supply_curve/supply_curve.py:268]: maybe add parameters/returns docstring?

File: reV/supply_curve/supply_curve.py [hunk 2]
```diff
@@ -612,61 +656,85 @@ def _compute_trans_cap_cost(trans_table, trans_costs=None,
...
+            Flag to only compute tranmission capital cost if transmission
+            feature has enough available capacity, by default True
+        max_workers : int | NoneType, optional
+            Number of workers to use to compute lcot, if > 1 run in parallel.
+            None uses all available cpu's. by default None
+        consider_friction : bool, optional
+            Flag to consider friction layer on LCOE, by default True
+        """
+        trans_costs = transmission_costs
+        self._trans_table = self._add_trans_lcot(self._trans_table, fcr,
```
Comment (grantbuster) [inline reV/supply_curve/supply_curve.py:723]: You should merge this method `_add_trans_lcot()` into the public method `compute_total_lcoe()`

File: reV/supply_curve/supply_curve.py [hunk 3]
```diff
@@ -612,61 +656,85 @@ def _compute_trans_cap_cost(trans_table, trans_costs=None,
...
+            Flag to have substation connection is limited by maximum capacity
+            of the attached lines, legacy method, by default False
+        connectable : bool, optional
+            Flag to only compute tranmission capital cost if transmission
+            feature has enough available capacity, by default True
+        max_workers : int | NoneType, optional
+            Number of workers to use to compute lcot, if > 1 run in parallel.
+            None uses all available cpu's. by default None
+        consider_friction : bool, optional
+            Flag to consider friction layer on LCOE, by default True
```
Comment (grantbuster) [inline reV/supply_curve/supply_curve.py:720]: Could we change this docstring to "Flag to consider friction layer on LCOE when "mean_lcoe_friction" is in the sc points input, by default True"

**Review by grantbuster — CHANGES_REQUESTED** [review]

File: reV/supply_curve/supply_curve.py (hunk 1 above)
Comment (MRossol) [reply reV/supply_curve/supply_curve.py:268]: Whoops, pulled those from a test file, fixed now

File: reV/supply_curve/supply_curve.py (hunk 2 above)
Comment (MRossol) [reply reV/supply_curve/supply_curve.py:723]: done

Comment (MRossol) [reply reV/supply_curve/supply_curve.py:723]: good call, done

**Review by grantbuster — APPROVED** [review]

**Pull request merged**